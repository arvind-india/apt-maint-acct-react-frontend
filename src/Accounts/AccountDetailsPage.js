import React from 'react'
import { connect } from 'react-redux'
import Select from 'react-select';

import {
    Form,
    Button,
    FormGroup,
    FormText,
    Input,
    Label,
} from 'reactstrap'

import { accountActions as actions,
         alertActions,
         flatActions
} from '../_actions'

import { CATEGORIES, MONTHS } from '../_constants'
import { FlashMessage } from '../_components'


let module = 'accounts' // module name

export class AccountDetails extends React.Component {

  constructor(props) {
    super(props)
    const { location } = props
    let model = location.state.model // model supplied from list page
    let initializeModel = {
      id: model.id,
      recorded_at: model.recorded_at,
      item: model.item,
      flat_number: model.flat_number,
      name: model.name,
      for_month: model.for_month,
      for_year: model.for_year,
      crdr: model.crdr,
      amount: model.amount,
      balance: model.balance,
      category: model.category,
      remarks: model.remarks
    }
    this.state = {
      model: initializeModel,           // model to edit
      submitted: false,
      adding: model.id === 0
    }
    this.handleChange = this.handleChange.bind(this)
    this.handleCategoryChange = this.handleCategoryChange.bind(this)
    this.handleFlatChange = this.handleFlatChange.bind(this)
    this.handleMonthChange = this.handleMonthChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)

    this.props.getAllFlats()
  }

  canBeSaved() { // check for changes in model, if changes present, it can save
    const { model } = this.state

    if (!model.recorded_at) return false
    if (!model.item) return false
    if (!model.name) return false
    if (!model.for_month) return false
    if (!model.for_year) return false
    if (!model.crdr) return false
    if (!model.amount) return false
    if (!model.category) return false
    if (!model.remarks) return false

    return true // can save changes
  }

  render() {
    const { alert } = this.props
    return (
      <div>
        <h2>Account Details</h2>
        {alert.message && <FlashMessage text={alert.message} color={alert.color} delay={2100}/>}
        { this.validateForm() }
        { this.show() }
      </div>
    )
  }

  show(){
    const { adding } = this.state
    const { authzn } = this.props
    let title = adding?'Add':authzn.allowsEdit?'Edit':'View'
    return <Form id="accountDetailsForm" onSubmit={this.handleSubmit} className="grid-form">
      <fieldset>
  			<legend>{title}</legend>
        <div data-row-span="2">
          {this.showRecordedAt()}
          {this.showItem()}
        </div>
        <div data-row-span="1">
          {this.showCategory()}
        </div>
        <div data-row-span="2">
          {this.showFlatNumber()}
          {this.showName()}
        </div>
        <div data-row-span="4">
          {this.showMonth()}
          {this.showYear()}
          {this.showAmount()}
          {this.showCrdr()}
        </div>
        <div data-row-span="1">
          {this.showRemarks()}
        </div>
      </fieldset>
      <br/>
      <Button
        type="submit"
        color="primary"
        disabled={!this.formValid}
        hidden={!authzn.allowsEdit}
        title={this.validationMsg}
        >Save</Button>
      <Button
        color="link"
        className="text-danger"
        title="Go back"
        onClick={() => this.props.history.goBack()}
      >Cancel</Button>
    </Form>
  }

  showRecordedAt() {
    const { submitted, model } = this.state
    return <div data-field-span="1">
				<Label>Txn Date</Label>
        <Input
          id="recorded_at"
          type="date"
          name="recorded_at"
          value={model.recorded_at}
          placeholder="Transaction Date here"
          className="inputField"
          onChange={this.handleChange}
        />
        {submitted && !model.recorded_at
          && <FormText color="danger">Transaction date is required</FormText>}
			</div>
  }

  handleChange(event) {
    const { name, value } = event.target
    const { model } = this.state
    this.setState({
        model: {
          ...model,
          [name]: value
        }
    })
  }

  showItem() {
    const { submitted, model } = this.state
    return <div data-field-span="1">
				<Label>Item</Label>
        <Input
          id="item"
          type="text"
          name="item"
          value={model.item}
          placeholder="Item details here"
          className="inputField"
          onChange={this.handleChange}
        />
        {submitted && !model.item
          && <FormText color="danger">Item is required</FormText>}
			</div>
  }

  handleSubmit(event) {
    const { model } = this.state
    event.preventDefault()
    this.setState({ submitted: true })
    this.props.saveChanges(model)
  }

  validateForm() {
    if ( this.changedProps().length === 0 ) {
      this.validationMsg = 'No changes to save'
      this.formValid = false
      return null
    }
    if( this.canBeSaved() ) {
      this.validationMsg = 'Save Changes'
      this.formValid = true
      return null
    }

    // finally, if reached here
    this.validationMsg = 'Missing "Required Data"...'
    this.formValid = false
  }

  changedProps() {
    const { model } = this.state
    const { location } = this.props
    let modelDB = location.state.model
    let props = []

    // check for changes in model props
    for(const prop in model) {
      if( prop === 'id') continue // exclude 'id' from comparision
      if(modelDB[prop] !== model[prop]) { // if data is changed wrt data in database
        props.push(prop)
      }
    }
    return props;
  }

  showCategory() {
    const { submitted, model } = this.state
    return <div data-field-span="1">
				<Label>Category</Label>
        <Select
          id="category"
          name="form-field-name"
          value={model.category}
          multi={false}
          joinValues={false}
          simpleValue={true}
          placeholder="Select Category..."
          onChange={this.handleCategoryChange}
          valueKey="name"
          labelKey="name"
          options={CATEGORIES}
        />
        {submitted && !model.category
          && <FormText color="danger">Category is required</FormText>}
			</div>
  }

  handleCategoryChange(selectedOption) {
    const { model } = this.state
    this.setState({
      model: {
        ...model,
        category: selectedOption
      }
    })
  }

  showFlatNumber() {
    const { flats } = this.props
    const { model } = this.state
    let options = flats.items?flats.items:[]

    return <div data-field-span="1">
				<Label>Flat Number (optional)</Label>
        <Select
          id="flat_number"
          name="form-field-name"
          value={model.flat_number}
          multi={false}
          joinValues={false}
          simpleValue={true}
          placeholder="Select a Flat Number..."
          onChange={this.handleFlatChange}
          valueKey="flat_number"
          labelKey="flat_number"
          options={options}
        />
			</div>
  }

  handleFlatChange(selectedOption) {
    const { model } = this.state
    this.setState({
      model: {
        ...model,
        flat_number: selectedOption
      }
    })
  }

  showName() {
    const { submitted, model } = this.state
    return <div data-field-span="1">
				<Label>Name</Label>
        <Input
          id="name"
          type="text"
          name="name"
          value={model.name}
          placeholder="Name here"
          className="inputField"
          onChange={this.handleChange}
        />
        {submitted && !model.name
          && <FormText color="danger">Name is required</FormText>}
			</div>
  }

  showMonth() {
    const { submitted, model } = this.state
    return <div data-field-span="1">
				<Label>Month</Label>
        <Select
          id="for_month"
          name="form-field-name"
          value={model.for_month}
          multi={false}
          joinValues={false}
          simpleValue={true}
          placeholder="Select a Month..."
          onChange={this.handleMonthChange}
          valueKey="number"
          labelKey="name"
          options={MONTHS}
        />
        {submitted && !model.for_month
          && <FormText color="danger">Month is required</FormText>}
			</div>
  }
  handleMonthChange(selectedOption) {
    const { model } = this.state
    this.setState({
      model: {
        ...model,
        for_month: selectedOption
      }
    })
  }

  showYear() {
    const { submitted, model } = this.state
    return <div data-field-span="1">
				<Label>Year</Label>
        <Input
          id="for_year"
          type="number"
          name="for_year"
          value={model.for_year}
          placeholder="Year here"
          className="inputField"
          min="2013"
          max="2030"
          onChange={this.handleChange}
        />
        {submitted && !model.for_year
          && <FormText color="danger">Year is required</FormText>}
			</div>
  }

  showAmount() {
    const { submitted, model } = this.state
    return <div data-field-span="1">
				<Label>Amount</Label>
        <Input
          id="amount"
          type="number"
          step="0.01"
          name="amount"
          value={model.amount}
          placeholder="Amount here"
          className="inputField"
          onChange={this.handleChange}
        />
        {submitted && !model.amount
          && <FormText color="danger">Amount is required</FormText>}
			</div>
  }

  showCrdr() {
    const { model } = this.state
    let rtype = model && model.crdr?model.crdr:''

    return <div data-field-span="1">
        <Label>Collection/Expenditure</Label>
        <FormGroup check inline>
          <Label check>
            <Input
              className="crdr"
              type="radio"
              name="crdr"
              value="cr"
              checked={rtype === "cr"}
              onChange={this.handleChange}
            /> Cr
          </Label>
        </FormGroup>
        <FormGroup check inline>
          <Label check>
            <Input
              className="crdr"
              type="radio"
              name="crdr"
              value="dr"
              checked={rtype === "dr"}
              onChange={this.handleChange}
            /> Dr
          </Label>
        </FormGroup>
      </div>
  }

  showRemarks() {
    const { submitted, model } = this.state
    return <div data-field-span="1">
				<Label>Remarks</Label>
        <Input
          id="remarks"
          type="text"
          name="remarks"
          value={model.remarks}
          placeholder="Remarks here"
          className="inputField"
          onChange={this.handleChange}
        />
        {submitted && !model.remarks
          && <FormText color="danger">Remarks is required</FormText>}
			</div>
  }

}

function mapStateToProps(state) {
  const { alert, authorizations, users, flats } = state
  const authzn = authorizations[module]
  return {
    alert,
    authzn,
    users,
    flats
  }
}

function mapDispatchToProps(dispatch) {
  return {
    getAllFlats: () => {
      dispatch(flatActions.getAll())
    },
    saveChanges: (model) => {
      dispatch(actions.saveChanges(model))
    },
    error: (msg) => {
      dispatch(alertActions.error(msg))
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(AccountDetails)
