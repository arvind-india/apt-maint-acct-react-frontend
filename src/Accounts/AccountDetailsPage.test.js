// AccountsPage.test.js0]

import React from 'react'
import { shallow } from 'enzyme'

import { AccountDetails } from './AccountDetailsPage'

const mockAccounts = [
  {
    id: 1,
    recorded_at: '2018-04-04',
    item: 'maintenance fee',
    flat_number: 'G1',
    name: 'tester',
    for_month: '04',
    for_year: '2018',
    crdr: 'cr',
    amount: '600',
    balance: '',
    category: 'maintenance',
    remarks: 'maintenance for the month april 2018'
  },
  {
    id: 2,
    recorded_at: '2018-04-01',
    item: 'sewage clearance',
    flat_number: 'NA',
    name: 'Sewage Truck Service',
    for_month: '04',
    for_year: '2018',
    crdr: 'dr',
    amount: '2200',
    balance: '',
    category: 'sewage disposal',
    remarks: 'septic tank clearance'
  }
]
const mockFlats = [
  {id: 1, flat_number: 'G1'},
  {id: 2, flat_number: 'G2'}
]

const mockProps = [
  {
    alert: {message: "testingAlert"},
    authzn: {
      allowsAdd: true,
      allowsEdit: true,
      allowsDelete: true,
      condition: ''
    },
    getAllFlats: jest.fn(),
//    clearAlert: jest.fn(),
    saveChanges: jest.fn(),
    error: jest.fn(),
    accounts: {
      items: mockAccounts
    },
    flats: {
      items: mockFlats
    }
  }
]
const mockLocations = [
  {
    state: {
      model: mockAccounts[0],
      submitted: false,
      adding: false
    }
  },
  {
    state: {
      model: mockAccounts[1],
      submitted: false,
      adding: false
    }
  }
]

const mocks = [
  {
    recorded_at: {name: 'recorded_at', value: mockAccounts[0].recorded_at},
    item: {name: 'item', value: mockAccounts[0].item},
    category: mockAccounts[0].category,
    flat_number: mockAccounts[0].flat_number,
    name: {name: 'name', value: mockAccounts[0].name},
    for_month: mockAccounts[0].for_month,
    for_year: {name: 'for_year', value: mockAccounts[0].for_year},
    crdr: {name: 'crdr', value: mockAccounts[0].crdr},
    amount: {name: 'amount', value: mockAccounts[0].amount},
    balance: {name: 'balance', value: mockAccounts[0].balance},

    remarks: {name: 'remarks', value: mockAccounts[0].remarks}
  }
]

const component = shallow(<AccountDetails {...mockProps[0]} location={mockLocations[0]}/>)

describe('AccountsPage params test', () => {
  it('should render AccountsPage without crashing', () => {
    component
  })
  it('should render initial layout', () => {
    expect(component.getElements()).toMatchSnapshot()
  })
  it('should have model state', () => {
    expect(component.state('model')).toBeDefined()
  })
  it('should have submitted state', () => {
    expect(component.state('submitted')).toBeDefined()
  })
  it('should have adding state', () => {
    expect(component.state('adding')).toBeDefined()
  })
})


describe('AccountDetailsPage events test 1', () => {
/*  it('should call clearAlert on submit and no crash', () => {
    component.find('#accountDetailsForm').simulate('submit', { preventDefault() {} })
    expect(mockProps[0].clearAlert.mock.calls.length).toEqual(1)
  }) */

  it('should be called with required states as arguments', () => {
    component.find('#recorded_at').simulate('change', {target: mocks[0].recorded_at})
    component.find('#item').simulate('change', {target: mocks[0].item})
    component.find('#flat_number').simulate('change', mocks[0].flat_number)
    component.find('#name').simulate('change', {target: mocks[0].name})
    component.find('#for_month').simulate('change', mocks[0].for_month)
    component.find('#for_year').simulate('change', {target: mocks[0].for_year})
    component.find('.crdr [checked=true]').simulate('change', {target: mocks[0].crdr})
    component.find('#amount').simulate('change', {target: mocks[0].amount})
    component.find('#category').simulate('change', mocks[0].category)
    component.find('#remarks').simulate('change', {target: mocks[0].remarks})
    component.find('#accountDetailsForm').simulate('submit', { preventDefault() {} })
    expect(mockProps[0].saveChanges.mock.calls[0][0]).toEqual(mockAccounts[0])
  })


})

/*
const component2 = shallow(<AccountDetails {...mockProps[1]} location={mockLocations[1]}/>)
describe('AccountDetailsPage events test 2', () => {
  it('should be called with required states as arguments', () => {
    component2.find('#blockNumber').simulate('change', {target: mocks[1].blockNumber})
    component2.find('#accountNumber').simulate('change', {target: mocks[1].accountNumber}) // calling again to toggle from '' to 'R'
    component2.find('#accountDetailsForm').simulate('submit', { preventDefault() {} })
    expect(mockProps[1].saveChanges.mock.calls[0][0]).toEqual(mockAccounts[1])
  })
})
*/
