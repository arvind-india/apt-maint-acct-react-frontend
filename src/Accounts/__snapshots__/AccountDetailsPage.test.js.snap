// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountsPage params test should render initial layout 1`] = `
Array [
  <div>
    <h2>
        Account Details
    </h2>
    <FlashMessage
        color={undefined}
        delay={2100}
        text="testingAlert"
    />
    <Form
        className="grid-form"
        id="accountDetailsForm"
        onSubmit={[Function]}
        tag="form"
    >
        <fieldset>
            <legend>
                Edit
            </legend>
            <div
                data-row-span="2"
            >
                <div
                    data-field-span="1"
                >
                    <Label
                        tag="label"
                        widths={
                            Array [
                                "xs",
                                "sm",
                                "md",
                                "lg",
                                "xl",
                              ]
                        }
                    >
                        Txn Date
                    </Label>
                    <Input
                        className="inputField"
                        id="recorded_at"
                        name="recorded_at"
                        onChange={[Function]}
                        placeholder="Transaction Date here"
                        tag="p"
                        type="date"
                        value="2018-04-04"
                    />
                </div>
                <div
                    data-field-span="1"
                >
                    <Label
                        tag="label"
                        widths={
                            Array [
                                "xs",
                                "sm",
                                "md",
                                "lg",
                                "xl",
                              ]
                        }
                    >
                        Item
                    </Label>
                    <Input
                        className="inputField"
                        id="item"
                        name="item"
                        onChange={[Function]}
                        placeholder="Item details here"
                        tag="p"
                        type="text"
                        value="maintenance fee"
                    />
                </div>
            </div>
            <div
                data-row-span="1"
            >
                <div
                    data-field-span="1"
                >
                    <Label
                        tag="label"
                        widths={
                            Array [
                                "xs",
                                "sm",
                                "md",
                                "lg",
                                "xl",
                              ]
                        }
                    >
                        Category
                    </Label>
                    <Select
                        arrowRenderer={[Function]}
                        autosize={true}
                        backspaceRemoves={true}
                        backspaceToRemoveMessage="Press backspace to remove {label}"
                        clearAllText="Clear all"
                        clearRenderer={[Function]}
                        clearValueText="Clear value"
                        clearable={true}
                        closeOnSelect={true}
                        deleteRemoves={true}
                        delimiter=","
                        disabled={false}
                        escapeClearsValue={true}
                        filterOptions={[Function]}
                        id="category"
                        ignoreAccents={true}
                        ignoreCase={true}
                        inputProps={Object {}}
                        isLoading={false}
                        joinValues={false}
                        labelKey="name"
                        matchPos="any"
                        matchProp="any"
                        menuBuffer={0}
                        menuRenderer={[Function]}
                        multi={false}
                        name="form-field-name"
                        noResultsText="No results found"
                        onBlurResetsInput={true}
                        onChange={[Function]}
                        onCloseResetsInput={true}
                        onSelectResetsInput={true}
                        openOnClick={true}
                        optionComponent={[Function]}
                        options={
                            Array [
                                Object {
                                  "name": "Electrical",
                                },
                                Object {
                                  "name": "Garbage",
                                },
                                Object {
                                  "name": "Monthly Maintenance",
                                },
                                Object {
                                  "name": "Major Maintenance",
                                },
                                Object {
                                  "name": "Plumbing",
                                },
                                Object {
                                  "name": "Seasonal Gift",
                                },
                                Object {
                                  "name": "Septic Tank",
                                },
                                Object {
                                  "name": "Sweeping",
                                },
                                Object {
                                  "name": "Water Tank - Overhead",
                                },
                                Object {
                                  "name": "Water Tank - Sump",
                                },
                                Object {
                                  "name": "Others",
                                },
                              ]
                        }
                        pageSize={5}
                        placeholder="Select Category..."
                        removeSelected={true}
                        required={false}
                        rtl={false}
                        scrollMenuIntoView={true}
                        searchable={true}
                        simpleValue={true}
                        tabSelectsValue={true}
                        trimFilter={true}
                        value="maintenance"
                        valueComponent={[Function]}
                        valueKey="name"
                    />
                </div>
            </div>
            <div
                data-row-span="2"
            >
                <div
                    data-field-span="1"
                >
                    <Label
                        tag="label"
                        widths={
                            Array [
                                "xs",
                                "sm",
                                "md",
                                "lg",
                                "xl",
                              ]
                        }
                    >
                        Flat Number (optional)
                    </Label>
                    <Select
                        arrowRenderer={[Function]}
                        autosize={true}
                        backspaceRemoves={true}
                        backspaceToRemoveMessage="Press backspace to remove {label}"
                        clearAllText="Clear all"
                        clearRenderer={[Function]}
                        clearValueText="Clear value"
                        clearable={true}
                        closeOnSelect={true}
                        deleteRemoves={true}
                        delimiter=","
                        disabled={false}
                        escapeClearsValue={true}
                        filterOptions={[Function]}
                        id="flat_number"
                        ignoreAccents={true}
                        ignoreCase={true}
                        inputProps={Object {}}
                        isLoading={false}
                        joinValues={false}
                        labelKey="flat_number"
                        matchPos="any"
                        matchProp="any"
                        menuBuffer={0}
                        menuRenderer={[Function]}
                        multi={false}
                        name="form-field-name"
                        noResultsText="No results found"
                        onBlurResetsInput={true}
                        onChange={[Function]}
                        onCloseResetsInput={true}
                        onSelectResetsInput={true}
                        openOnClick={true}
                        optionComponent={[Function]}
                        options={
                            Array [
                                Object {
                                  "flat_number": "G1",
                                  "id": 1,
                                },
                                Object {
                                  "flat_number": "G2",
                                  "id": 2,
                                },
                              ]
                        }
                        pageSize={5}
                        placeholder="Select a Flat Number..."
                        removeSelected={true}
                        required={false}
                        rtl={false}
                        scrollMenuIntoView={true}
                        searchable={true}
                        simpleValue={true}
                        tabSelectsValue={true}
                        trimFilter={true}
                        value="G1"
                        valueComponent={[Function]}
                        valueKey="flat_number"
                    />
                </div>
                <div
                    data-field-span="1"
                >
                    <Label
                        tag="label"
                        widths={
                            Array [
                                "xs",
                                "sm",
                                "md",
                                "lg",
                                "xl",
                              ]
                        }
                    >
                        Name
                    </Label>
                    <Input
                        className="inputField"
                        id="name"
                        name="name"
                        onChange={[Function]}
                        placeholder="Name here"
                        tag="p"
                        type="text"
                        value="tester"
                    />
                </div>
            </div>
            <div
                data-row-span="4"
            >
                <div
                    data-field-span="1"
                >
                    <Label
                        tag="label"
                        widths={
                            Array [
                                "xs",
                                "sm",
                                "md",
                                "lg",
                                "xl",
                              ]
                        }
                    >
                        Month
                    </Label>
                    <Select
                        arrowRenderer={[Function]}
                        autosize={true}
                        backspaceRemoves={true}
                        backspaceToRemoveMessage="Press backspace to remove {label}"
                        clearAllText="Clear all"
                        clearRenderer={[Function]}
                        clearValueText="Clear value"
                        clearable={true}
                        closeOnSelect={true}
                        deleteRemoves={true}
                        delimiter=","
                        disabled={false}
                        escapeClearsValue={true}
                        filterOptions={[Function]}
                        id="for_month"
                        ignoreAccents={true}
                        ignoreCase={true}
                        inputProps={Object {}}
                        isLoading={false}
                        joinValues={false}
                        labelKey="name"
                        matchPos="any"
                        matchProp="any"
                        menuBuffer={0}
                        menuRenderer={[Function]}
                        multi={false}
                        name="form-field-name"
                        noResultsText="No results found"
                        onBlurResetsInput={true}
                        onChange={[Function]}
                        onCloseResetsInput={true}
                        onSelectResetsInput={true}
                        openOnClick={true}
                        optionComponent={[Function]}
                        options={
                            Array [
                                Object {
                                  "name": "Jan",
                                  "number": 1,
                                },
                                Object {
                                  "name": "Feb",
                                  "number": 2,
                                },
                                Object {
                                  "name": "Mar",
                                  "number": 3,
                                },
                                Object {
                                  "name": "Apr",
                                  "number": 4,
                                },
                                Object {
                                  "name": "May",
                                  "number": 5,
                                },
                                Object {
                                  "name": "Jun",
                                  "number": 6,
                                },
                                Object {
                                  "name": "Jul",
                                  "number": 7,
                                },
                                Object {
                                  "name": "Aug",
                                  "number": 8,
                                },
                                Object {
                                  "name": "Sep",
                                  "number": 9,
                                },
                                Object {
                                  "name": "Oct",
                                  "number": 10,
                                },
                                Object {
                                  "name": "Nov",
                                  "number": 11,
                                },
                                Object {
                                  "name": "Dec",
                                  "number": 12,
                                },
                              ]
                        }
                        pageSize={5}
                        placeholder="Select a Month..."
                        removeSelected={true}
                        required={false}
                        rtl={false}
                        scrollMenuIntoView={true}
                        searchable={true}
                        simpleValue={true}
                        tabSelectsValue={true}
                        trimFilter={true}
                        value="04"
                        valueComponent={[Function]}
                        valueKey="number"
                    />
                </div>
                <div
                    data-field-span="1"
                >
                    <Label
                        tag="label"
                        widths={
                            Array [
                                "xs",
                                "sm",
                                "md",
                                "lg",
                                "xl",
                              ]
                        }
                    >
                        Year
                    </Label>
                    <Input
                        className="inputField"
                        id="for_year"
                        max="2030"
                        min="2013"
                        name="for_year"
                        onChange={[Function]}
                        placeholder="Year here"
                        tag="p"
                        type="number"
                        value="2018"
                    />
                </div>
                <div
                    data-field-span="1"
                >
                    <Label
                        tag="label"
                        widths={
                            Array [
                                "xs",
                                "sm",
                                "md",
                                "lg",
                                "xl",
                              ]
                        }
                    >
                        Amount
                    </Label>
                    <Input
                        className="inputField"
                        id="amount"
                        name="amount"
                        onChange={[Function]}
                        placeholder="Amount here"
                        step="0.01"
                        tag="p"
                        type="number"
                        value="600"
                    />
                </div>
                <div
                    data-field-span="1"
                >
                    <Label
                        tag="label"
                        widths={
                            Array [
                                "xs",
                                "sm",
                                "md",
                                "lg",
                                "xl",
                              ]
                        }
                    >
                        Collection/Expenditure
                    </Label>
                    <FormGroup
                        check={true}
                        inline={true}
                        tag="div"
                    >
                        <Label
                            check={true}
                            tag="label"
                            widths={
                                Array [
                                    "xs",
                                    "sm",
                                    "md",
                                    "lg",
                                    "xl",
                                  ]
                            }
                        >
                            <Input
                                checked={true}
                                className="crdr"
                                name="crdr"
                                onChange={[Function]}
                                tag="p"
                                type="radio"
                                value="cr"
                            />
                             Cr
                        </Label>
                    </FormGroup>
                    <FormGroup
                        check={true}
                        inline={true}
                        tag="div"
                    >
                        <Label
                            check={true}
                            tag="label"
                            widths={
                                Array [
                                    "xs",
                                    "sm",
                                    "md",
                                    "lg",
                                    "xl",
                                  ]
                            }
                        >
                            <Input
                                checked={false}
                                className="crdr"
                                name="crdr"
                                onChange={[Function]}
                                tag="p"
                                type="radio"
                                value="dr"
                            />
                             Dr
                        </Label>
                    </FormGroup>
                </div>
            </div>
            <div
                data-row-span="1"
            >
                <div
                    data-field-span="1"
                >
                    <Label
                        tag="label"
                        widths={
                            Array [
                                "xs",
                                "sm",
                                "md",
                                "lg",
                                "xl",
                              ]
                        }
                    >
                        Remarks
                    </Label>
                    <Input
                        className="inputField"
                        id="remarks"
                        name="remarks"
                        onChange={[Function]}
                        placeholder="Remarks here"
                        tag="p"
                        type="text"
                        value="maintenance for the month april 2018"
                    />
                </div>
            </div>
        </fieldset>
        <br />
        <Button
            color="primary"
            disabled={true}
            hidden={false}
            tag="button"
            title="No changes to save"
            type="submit"
        >
            Save
        </Button>
        <Button
            className="text-danger"
            color="link"
            onClick={[Function]}
            tag="button"
            title="Go back"
        >
            Cancel
        </Button>
    </Form>
</div>,
]
`;
